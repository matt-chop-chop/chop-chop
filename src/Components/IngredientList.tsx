import { Box, Flex, Heading, Image, Text } from "@chakra-ui/react";

type IngredientListProps = {
  ingredients: RecipeIngredient[];
};

const IngredientList = ({ ingredients }: IngredientListProps) => {
  if (!ingredients.length) return null;

  return (
    <Box my={5}>
      <Heading as="h2" mb={4} variant="heading-medium">
        Ingredients
      </Heading>
      {ingredients.map((ingredient, index) => (
        <Flex key={index} width="100%">
          <Text ml={[4, 6, 8]} mt={2} variant="body-large">
            â€¢ {ingredient.measurement} {ingredient.name}
          </Text>
          {ingredient.image && (
            <Image
              alt="An ingredient image"
              maxHeight="50px"
              ml={4}
              src={ingredient.image}
            />
          )}
        </Flex>
      ))}
    </Box>
  );
};

export default IngredientList;
