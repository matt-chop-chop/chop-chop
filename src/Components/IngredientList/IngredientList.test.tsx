import { shallow } from "enzyme";
import React from "react";
import IngredientList, { IngredientListProps } from "./IngredientList";

const props: IngredientListProps = {
  ingredients: [
    { name: "Test Ingredient", measurement: "500g", image: "/test" },
    { name: "Test Ingredient 2", measurement: "500g", image: "/test" },
  ],
};

describe("IngredientList", () => {
  it("is exported from IngredientList", () => {
    expect(IngredientList).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<IngredientList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("returns null if the list of ingredients is empty ", () => {
    const wrapper = shallow(<IngredientList ingredients={[]} />);

    expect(wrapper.type()).toBeNull();
  });
});
