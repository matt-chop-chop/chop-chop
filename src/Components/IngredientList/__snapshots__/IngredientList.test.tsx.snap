// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientList renders as expected  1`] = `
<Box
  my={5}
>
  <Heading
    as="h2"
    mb={4}
    variant="heading-medium"
  >
    Ingredients
  </Heading>
  <Flex
    key="0"
    width="100%"
  >
    <Text
      ml={
        [
          4,
          6,
          8,
        ]
      }
      mt={2}
      variant="body-large"
    >
      • 
      500g
       
      Test Ingredient
    </Text>
    <Image
      alt="An ingredient image"
      maxHeight="50px"
      ml={4}
      src="/test"
    />
  </Flex>
  <Flex
    key="1"
    width="100%"
  >
    <Text
      ml={
        [
          4,
          6,
          8,
        ]
      }
      mt={2}
      variant="body-large"
    >
      • 
      500g
       
      Test Ingredient 2
    </Text>
    <Image
      alt="An ingredient image"
      maxHeight="50px"
      ml={4}
      src="/test"
    />
  </Flex>
</Box>
`;
